### PR TITLE
New package: simulide-0.4.14

### PR DESCRIPTION
**Pull request by Logarithmus**
Real time electronic circuit simulator with PIC, AVR and Arduino simulation.
It bundles some older versions of `simavr` and `gpsim`. I tried to make it use up-to-date versions, which are installed system-wide, but turns out there's plenty of fixing needed to do it. So I postponed it for the future.
P. S. I'm not the author of this tool, I just use it for my university MCU programming course.

### CODE REVIEW

File: srcpkgs/simulide/patches/00-glib.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ INCLUDEPATH += ../src \
+     ../src/gui \
+@@ -140,7 +140,9 @@
+     ../src/gpsim \
+     ../src/gpsim/devices \
+     ../src/gpsim/modules \
+-    ../src/gpsim/registers
++    ../src/gpsim/registers \
++     /usr/include/glib-2.0 \
++     /usr/lib/glib-2.0/include
```
Comment (ericonr) [inline srcpkgs/simulide/patches/00-glib.patch:21]: I'm pretty sure this breaks cross compilation.

File: srcpkgs/gpsim/template [hunk 2]
```diff
@@ -1,27 +0,0 @@
-# Template file for 'gpsim'
```
Comment (ericonr) [inline srcpkgs/gpsim/template:1]: Why was this removed?

Comment (Logarithmus) [reply srcpkgs/gpsim/template]: As I said, I was trying to use `gpsim` and `simavr` version provided by Void repo.
By a mistake, I removed `gpsim` entirely instead of reverting to `master`.
TLDR; Fixed

File: srcpkgs/simulide/patches/00-glib.patch (hunk 1 above)
Comment (Logarithmus) [reply srcpkgs/simulide/patches/00-glib.patch:21]: I'd like to get some advise on handling this.

Comment (sgn) [reply srcpkgs/simulide/patches/00-glib.patch:21]: I guess this chunk (in template file) would work (not tested):
```sh
CXXFLAGS="$($PKG_CONFIG --cflags glib-2.0)"
```

Comment (sgn) [reply srcpkgs/simulide/patches/00-glib.patch:21]: With that change added, I don't think we need to patch.

File: srcpkgs/simulide/template [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+hostmakedepends="qt5-qmake pkg-config"
+makedepends="qt5-devel qt5-svg-devel qt5-multimedia-devel qt5-serialport-devel
+	qt5-script-devel elfutils-devel"
+short_desc="Electronic circuit simulator for PIC, AVR, Arduino"
+maintainer="Artur Sinila <opensource@logarithmus.dev>"
+license="GPL-3.0-or-later"
+homepage="https://simulide.com"
+distfiles="https://mailfence.com/pub/docs/santigoro/web/${_pkgname}_${version}/${_pkgname}_${version}-${_rev}_Sources.tar.gz"
+checksum=b55b03e507882b8b68a25e5155dcac6020ba273be93bdf689612a0be67056d43
+nopie_files="/usr/bin/simulide"
```
Comment (the-maldridge) [inline srcpkgs/simulide/template:19]: Generally we prefer to build all binaries with PIE flags, and needing to set this here is generally an indication that the xbps-src CFLAGS are not being passed to the build.  Can you check into this?

Comment (Logarithmus) [reply srcpkgs/simulide/template:19]: A few days later

Comment (Logarithmus) [reply srcpkgs/simulide/template]: Finally I've got some time for this. Now PIC & PIE are enabled.

**Comment by Logarithmus** [comment]
@ericonr ping
